auto_nag/scripts/crash_small_volume.py

```diff
@@ -0,0 +1,219 @@
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+
+
+class CrashSmallVolume(BzCleaner):
+    def __init__(self, min_crash_volume: int = 5):
+        """Constructor.
+
+        Args:
+            min_crash_volume: the minimum number of crashes per week for a
+                signature to not be considered low volume.
+        """
+        super().__init__()
+
+        self.min_crash_volume = min_crash_volume
+        topcrash = Topcrash(
+            criteria=self._adjust_topcrash_criteria(TOP_CRASH_IDENTIFICATION_CRITERIA)
+        )
+        self.topcrash_signatures = topcrash.get_signatures()
+        self.blocked_signatures = topcrash.get_blocked_signatures()
+
+    def description(self):
+        return "Bugs with small crash volume"
+
+    def columns(self):
+        return ["id", "summary", "severity", "deleted_keywords_count"]
+
+    def _get_last_topcrash_added(self, bug):
+        pass
+
+    def _adjust_topcrash_criteria(self, topcrash_criteria):
+        factor = 2
+        new_criteria = []
+        for criterion in topcrash_criteria:
+            criterion = {
+                **criterion,
+                "tc_limit": criterion["tc_limit"] * factor,
+            }
+            if "tc_startup_limit" in criterion:
+                criterion["tc_startup_limit"] = criterion["tc_startup_limit"] * factor
+
+            new_criteria.append(criterion)
+
+        return new_criteria
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+
+        if any(signature in self.blocked_signatures for signature in signatures):
+            # Ignore those bugs as we can't be sure.
+            return None
+
+        top_crash_signatures = [
+            signature
+            for signature in signatures
+            if signature in self.topcrash_signatures
+        ]
+
+        keep_topcrash_startup = any(
+            any(
+                criterion["is_startup"]
+                for criterion in self.topcrash_signatures[signature]
+            )
+            for signature in top_crash_signatures
+        )
+
+        keywords_to_remove = None
+        if not top_crash_signatures:
+            keywords_to_remove = set(bug["keywords"]) & {"topcrash", "topcrash-startup"}
+        elif not keep_topcrash_startup:
+            keywords_to_remove = set(bug["keywords"]) & {"topcrash-startup"}
+        else:
+            return None
+
+        data[bugid] = {
+            "severity": bug["severity"],
+            "ignore_severity": (
+                bug["severity"] not in HIGH_SEVERITY
+                or bug["groups"] == "security"
+                or any(keyword.startswith("sec-") for keyword in bug["keywords"])
+                or self._has_severity_downgrade_comment(bug)
+            ),
+            "keywords_to_remove": keywords_to_remove,
```

Comment:
Nit: to keep all the logic in the same place and avoid having to scroll back and forth, what about moving these to `set_autofix`?

Reply:
The downside of this is that we will need to keep `comments`, `keywords`, and `groups` for all bugs. Then, we need to do the same logical decision in multiple places (unless we set the `ignore_severity` field inside `set_autofix`). I prefer as possible not to mutate the bug object outside the `handle_bug` method.